hw/top_earlgrey/data/chip_testplan.hjson

```diff
@@ -283,16 +292,48 @@
       name: chip_usb_suspend
       desc: '''Verify that the USB device can detect the presence of VBUS from the USB host.
 
-            TBD.
+            - On the top of `chip_usb_fs_df_tx_rx`, add below at the end of the sequence.
+            - Configure USB device to enter `Suspend` state and ensure `link_suspend` interrupt is
+              triggered.
+            - Test these 2 power modes
+              - Only USB is in power-off, the other blocks are still powered.
```

Comment:
just to clarify, do you refer to the USB IP here or the USB link when mentioning "power-off"?
because AFAIK, the power to USB cannot be turned off without powering off all other modules that are not AON. the only thing we can do is clock gate it individually, I think (but it will still be powered on). CC @tjaychen

Reply:
Thanks for pointing this out.
I think 2 power modes may refer to normal sleep and deep sleep, but I'm not sure what we actually turn off in USB when we enter these 2 modes. And USB is also in always-on.

@tjaychen how can we test in this 2 power modes?

Reply:
so the basic idea is no matter what low power state we are in, normal or deep, the usb suspend / resume should work correctly.

In the normal sleep case, only the clocks are disabled but the power is kept on.  However, a resume condition should still be able to come through pinmux, cause pwrmgr to resume clocks, and for usbdev to correctly operate again.

In the case of deep sleep, the power is turned off the usbdev itself is in reset. When a resume condition comes, it will go through pinmux and cause pwrmgr to power on usbdev and everything else.  In this case, usb has been reset, so all of its previous state was lost.  Then the top level test either needs to re-enumerate, or restore previously enumerated information to continue operation (it is assumed the info would be stored somewhere always on like retention ram).  This testing obviously needs A LOT more software coordination.

The other case is when usb goes into suspend, but the system is still awake.  The wakeup then doesn't need to go through pinmux, but should hit usb directly and have everything resume.

So once these separate / static cases are tested, the interesting cases are when usb goes to suspend during awake state, the system THEN goes into one of the low power states.  We have to make sure that works as well.

Does that help? I can help document this somewhere to make it clear.

Reply:
Thanks @tjaychen

I just updated this part in testplan. Please take a look.